From 6a209c77c3f2d844eb06cfa2988ed1fb9357d20f Mon Sep 17 00:00:00 2001
From: Avra Sengupta <asengupt@redhat.com>
Date: Tue, 19 Jan 2016 16:22:17 +0530
Subject: [PATCH 249/251] snapshot: Return before redundant quorum check

    Backport of http://review.gluster.org/#/c/13260/

As of today, we don't support creation of snapshot even
if one brick is down. Hence the older quorum check is
redundant. Returning after performing the check to
see if all bricks are up.

Change-Id: I35661d05a15be0109aaae51b4fe0d5a8ca4333ad
BUG: 1299799
Signed-off-by: Avra Sengupta <asengupt@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/65966
Reviewed-by: Rafi Kavungal Chundattu Parambil <rkavunga@redhat.com>
Reviewed-by: Rajesh Joseph <rjoseph@redhat.com>
Tested-by: Rajesh Joseph <rjoseph@redhat.com>
---
 .../mgmt/glusterd/src/glusterd-snapshot-utils.c    |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/xlators/mgmt/glusterd/src/glusterd-snapshot-utils.c b/xlators/mgmt/glusterd/src/glusterd-snapshot-utils.c
index d45acbc..61a8ef9 100644
--- a/xlators/mgmt/glusterd/src/glusterd-snapshot-utils.c
+++ b/xlators/mgmt/glusterd/src/glusterd-snapshot-utils.c
@@ -2753,6 +2753,9 @@ glusterd_volume_quorum_calculate (glusterd_volinfo_t *volinfo, dict_t *dict,
                 *op_errstr = gf_strdup (err_str);
                 *op_errno = EG_BRCKDWN;
                 goto out;
+        } else {
+                quorum_met = _gf_true;
+                goto out;
         }
 
         up_count = volinfo->dist_leaf_count - down_count;
-- 
1.7.1

